rotkehlchen/chain/evm/decoding/superchain_bridge/decoder.py

```diff
@@ -95,16 +118,20 @@ def _decode_receive_deposit(self, context: DecoderContext) -> DecodingOutput:
                     asset=asset,
                     expected_event_type=expected_event_type,
                     new_event_type=new_event_type,
-                    counterparty=OPTIMISM_CPT_DETAILS,
+                    counterparty=self.counterparty,
                 )
 
         return DEFAULT_DECODING_OUTPUT
 
     # -- DecoderInterface methods
 
     def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
-        return dict.fromkeys(BRIDGE_ADDRESSES, (self._decode_receive_deposit,))
+        return dict.fromkeys(self.bride_addresses, (self._decode_receive_or_deposit,))
 
     @staticmethod
     def counterparties() -> tuple[CounterpartyDetails, ...]:
-        return (OPTIMISM_CPT_DETAILS,)
+        """Get counterparties for this bridge decoder.
```

Comment:
Do you then even have to implement it? Declare this as abstract class too and you don't have to. This should be an absract class since nobody should initiate an object of `SuperchainBridgeCommonDecoder` alone.